e2e-identity/tests/utils/docker/ldap.rs

```diff
@@ -55,47 +58,42 @@ impl LdapImage {
     }
 }
 
-#[derive(Debug, Default, Clone)]
-pub struct LdapArgs;
-
-impl ImageArgs for LdapArgs {
-    fn into_iterator(self) -> Box<dyn Iterator<Item = String>> {
-        Box::new(
-            vec![
-                "--copy-service".to_string(),
-                "--loglevel".to_string(),
-                "debug".to_string(),
-            ]
-            .into_iter(),
-        )
-    }
-}
-
 impl Image for LdapImage {
-    type Args = LdapArgs;
-
-    fn name(&self) -> String {
-        Self::NAME.to_string()
+    fn name(&self) -> &str {
+        Self::NAME
     }
 
-    fn tag(&self) -> String {
-        std::env::var("OPENLDAP_VERSION").unwrap_or_else(|_| Self::TAG.to_string())
+    fn tag(&self) -> &str {
+        std::env::var("OPENLDAP_VERSION")
+            .map(|tag| Box::leak(tag.into_boxed_str()) as &str)
+            .unwrap_or_else(|_| Self::TAG)
     }
 
     fn ready_conditions(&self) -> Vec<WaitFor> {
         vec![WaitFor::message_on_stderr("slapd starting")]
     }
 
-    fn env_vars(&self) -> Box<dyn Iterator<Item = (&String, &String)> + '_> {
+    fn env_vars(&self) -> impl IntoIterator<Item = (impl Into<Cow<'_, str>>, impl Into<Cow<'_, str>>)> {
         Box::new(self.env_vars.iter())
     }
 
-    fn volumes(&self) -> Box<dyn Iterator<Item = (&String, &String)> + '_> {
-        Box::new(self.volumes.iter())
+    fn mounts(&self) -> impl IntoIterator<Item = &Mount> {
+        &self.volumes
+    }
+
+    fn cmd(&self) -> impl IntoIterator<Item = impl Into<Cow<'_, str>>> {
```

Comment:
Shouldn't this be `impl Iterator`?? Given that you call `into_iter` below?

Reply:
This overrides a default trait implementation, so the whole function signature must match